src/ws/mod.rs

```diff
@@ -169,6 +192,7 @@ impl_ws_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T1
 pub struct WebSocketUpgrade<F, B, T> {
     callback: F,
     config: WebSocketConfig,
+    protocols: &'static [&'static str],
```

Comment:
I'm thinking if requiring `'static` is perhaps too restrictive here 🤔 

I'm wondering if we could make the struct field `Arc<[Cow<'static, str>]>`. That way its still cheap to clone and supports both string literals and `String`s. Using `Arc<[_]>` should be fine since we don't never have to mutate the list.

I think the builder method on `WebSocketUpgrade` can then be defined like this:

```rust
fn protocols(self, protocols: I) -> Self
where
    I: IntoIterator,
    I::Item: Into<Cow<'static, str>>,
{
    let protocols = protocols
        .into_iter()
        .map(Into::into)
        .collect::<Vec<_>>()
        .into();

    // ...
}
```

I feel that gives pretty good ergonomics while being light weight.

What do you think?

Reply:
You are right, this is more flexible. 🙂